pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4271,7 +4263,7 @@ private CompletableFuture<Topic> topicNotFoundReasonAsync(TopicName topicName) {
                                 == null ? "has no metadata" : "has zero partitions";
                         throw new RestException(Status.NOT_FOUND, String.format(
                                 "Partitioned Topic not found: %s %s", topicName.toString(), topicErrorType));
-                    } else if (!internalGetList(Optional.empty()).contains(topicName.toString())) {
+                    } else if (!internalGetListAsync(Optional.empty()).join().contains(topicName.toString())) {
```

Comment:
We need to avoid using the `join()` and `get()`.

Reply:
Thanks @nodece , I removed `join`  in `topicNotFoundReasonAsync`. PTAL